### ISSUE TITLE
MapDatasetOnOff Conversion - Issues with Definition of Alpha 

### ISSUE THREAD

**Issue by matthew-w-lundy** [gammapy/gammapy#5273]
**Gammapy version**
1.3.dev207+g751c350fd.d20240501

**Bug description**
Alpha in the MapDatasetOnOff is calculated as the ratio of acceptance. Then this is  brought over when you convert to a spectrum dataset using the `.to_spectrum_dataset() ` function. This results is a wrong definition of alpha with other implications. 

**Expected behavior**
I expect alpha to be equal to not only the ratio of exposures but also the ratio of the areas of the On/Off region convolved with the background kernel when appropriate. Since the alpha does not appear to be recalculated with the updated on region acceptance the returned alpha is that of a bin and not the alpha of the on/off region. This results in non-physical quantities.

**To Reproduce**
Using the RBM tutorial notebook change the following cell:
```
energy_axis_true = analysis.datasets[0].exposure.geom.axes["energy_true"]
stacked_on_off = MapDatasetOnOff.create(
    geom=geom_image, energy_axis_true=energy_axis_true, name="stacked"
)

for dataset in analysis.datasets:
    # Ring extracting makes sense only for 2D analysis
    dataset_on_off = ring_maker.run(dataset.to_image())
    stacked_on_off.stack(dataset_on_off)
```

to
```
from gammapy.datasets import Datasets
energy_axis_true = analysis.datasets[0].exposure.geom.axes["energy_true"]
stacked_on_off = MapDatasetOnOff.create(
    geom=geom_image, energy_axis_true=energy_axis_true, name="stacked"
)
stacked_on_off= Datasets()
for dataset in analysis.datasets:
    # Ring extracting makes sense only for 2D analysis
    dataset_on_off = ring_maker.run(dataset.to_image())
    #stacked_on_off.stack(dataset_on_off)
    dataset_on_off = dataset_on_off.to_spectrum_dataset(CircleSkyRegion(center=source_pos, radius=0.1 * u.deg), containment_correction=True)
    stacked_on_off.append(dataset_on_off)
stacked_on_off = stacked_on_off.stack_reduce()
display(stacked_on_off.info_dict())
```
**Other info**

I have had now ~3 new gammapy users move from the spectrum to the Ring Background tutorial and accidentally to use something similar to this to report significance, alpha, on and off counts so it's a fairly common issue. If you just adjust the acceptance there are a couple knock on implications that also need to be corrected.

**Comment by AtreyeeS** [comment]
Was this stacking issue resolved with #5270 @registerrier ?

I dont know if the use case reported here is completely safe. You are creating a `SpectrumDatasetOnOff` with a RingBackground. This is not the expected usage, and usage of the RingBackGround is traditionally not used for spectral analysis. Also, you are stacking a `SpectrumDatasetOnOff` onto a `MapDatasetOnOff`, which can be problematic.

**Comment by Astro-Kirsty** [comment]
What is the current status of this issue? @registerrier @AtreyeeS @matthew-w-lundy

**Comment by AtreyeeS** [comment]
Is this resolved now @samanthalwong ?

**Comment by samanthalwong** [comment]
@Astro-Kirsty @AtreyeeS, yes it is resolved! We are no longer using `dataset_on_off.to_spectrum_dataset` but running `ExcessMapEstimator` on the stacked dataset is working well to calculate counts, significance, etc.

**Comment by AtreyeeS** [comment]
Thanks! Closing then
